app/src/main/java/org/simple/clinic/user/UserSession.kt

```diff
@@ -76,6 +76,7 @@ class UserSession @Inject constructor(
     @Named("last_prescription_pull_token") private val prescriptionSyncPullToken: Preference<Optional<String>>,
     @Named("last_appointment_pull_token") private val appointmentSyncPullToken: Preference<Optional<String>>,
     @Named("last_medicalhistory_pull_token") private val medicalHistorySyncPullToken: Preference<Optional<String>>,
+    @Named("last_encounter_pull_token") private val encounterSyncPullToken: Preference<Optional<String>>,
```

Comment:
Curious, why do we have different pull tokens for different sync mechanisms?

Reply:
We have one token for every sync API. We cannot guarantee that all APIs will get synced in one go.
Hence we maintain different tokens to indicate what was last synced timestamp for each one so that the server can send the response accordingly.